numpy/linalg/linalg.py

```diff
@@ -543,6 +543,8 @@ def inv(a):
     with errstate(call=_raise_linalgerror_singular, invalid='call',
```

Comment:
Isn't `_raise_linalgerror_singular` supposed to detect this already? I worry that this is going to make matrix inversion much more expensive.

Reply:
I executed a sample code to check the inv() function and it is resulting in an incorrect output
![Test](https://github.com/numpy/numpy/assets/139049460/d01c510b-d77c-45c0-89a8-f1cd4a217185)

Reply:
`_raise_linalgerror_singular` isn't doing any checking; it's just transforming the generic exception from the `gufunc` machinery (and in turn from the underlying LAPACK routine) into the informative `LinAlgError`.